lib/common/options.js

```diff
@@ -209,6 +209,14 @@ Options.from = function (_options) {
     delete options.render.gzip
   }
 
+  // Legacy: Override build.htmlMinify with generate.minify if present
+  if (this.options.generate.minify) {
+    this.options.build.htmlMinify = this.options.generate.minify
```

Comment:
This part still have the issue we just discussed, I think it should only be executed in generation

Reply:
But how can we distinguish between `generated` and "normal" SSR?

Reply:
Unfortunately we can’t, so maybe just warning default value willl be applied but not override build options? @pi0 How do you think?

Reply:
That would be a SOFT BREAKING CHANGE :D Breaking change but warn to fix it :D But i agree.

Reply:
That's why I was unsure if we should really proceed this way. Also because can't oversee how much damage the new "defaults" can cause for SSR HTML minification :thinking: